src/System.Reflection.Emit.Lightweight/tests/DynamicILInfoTests.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    public class DynamicILInfoTests
+    {
+        private const string LocalSignature = "LocalSignature";
+        private const string Exceptions = "Exceptions";
+        private const string Code = "Code";
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public unsafe void SetX_ArrayNull_SetsToEmptyArray(bool skipVisibility)
+        {
+            DynamicMethod dynamicMethod = GetDynamicMethod(skipVisibility);
+            DynamicILInfo dynamicILInfo = dynamicMethod.GetDynamicILInfo();
+
+            dynamicILInfo.SetLocalSignature(null);
+            Assert.Equal(0, GetPropertyValue(dynamicILInfo, LocalSignature).Length);
```

Comment:
This is verifying internal implementation details via internal properties. Should this be rather verifying the publicly observable behavior with these settings?

Reply:
There are some more sophisticated tests in the DynamicILInfoGetTokenForTests class, which verify the publicly observable behavior and show usage.
btw, all the new tests I added are all cleaned up versions of tests in full framework.

Reply:
I do not think that these tests that check the internal state via private reflection are useful here. I would remove them.